src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -78,14 +78,20 @@ public FormulaValue GetField(string fieldName)
             return GetField(fieldType, fieldName);
         }
 
+        // Create an exception object for when the host violates the TryGetField() contract. 
+        private Exception HostException(string fieldName, string message)
+        {
+            return new InvalidOperationException($"{GetType().Name}.TryGetField({fieldName}): {message}");
```

Comment:
This will format the error message with a field value instead of a name, right?

Reply:
The error message tells the fieldname that was being accessed. Most importantly, it tells the derived type (which is likely where the bug is).